src/index.ts

```diff
@@ -37,32 +42,29 @@ export const checkTranslations = (
   options: Options = { format: "icu", checks: ["invalidKeys", "missingKeys"] }
 ): {
   missingKeys: CheckResult | undefined;
-  invalidKeys: CheckResult | undefined;
+  invalidKeys: InvalidTranslationsResult | undefined;
 } => {
   const { checks = ["invalidKeys", "missingKeys"] } = options;
 
-  let missingKeys = {};
-  let invalidKeys = {};
+  let missingKeys: CheckResult = {};
+  let invalidKeys: InvalidTranslationsResult = {};
 
   const hasMissingKeys = checks.includes("missingKeys");
   const hasInvalidKeys = checks.includes("invalidKeys");
 
   source.forEach(({ name, content }) => {
-    const files = targets
-      .filter(({ reference }) => reference === name)
-      .reduce((obj, { name: key, content }) => {
-        return Object.assign(obj, { [key]: content });
-      }, {});
+    const files = Object.fromEntries(
+      targets
+        .filter(({ reference }) => reference === name)
+        .map(({ name, content }) => [name, content])
+    );
 
     if (hasMissingKeys) {
-      Object.assign(missingKeys, checkMissingTranslations(content, files));
+      merge(missingKeys, checkMissingTranslations(content, files));
```

Comment:
This is also unrelated to the type issues but don't we need to merge these results? Same below.

Reply:
Isn't this what is happening here via the object.assign?

Reply:
Object.assign will just overwrite any existing values.

```js
Object.assign({a: ["foo"]}, {a: ["bar", "baz"]})
// => {a: ["bar", "baz"]}
```

Reply:
I think the scenario here is that the keys should be unique, but `merge` might be better.

```ts
Object.assign({a: ["foo"]}, {b: ["bar", "baz"]})
// => { a: [ 'foo' ], b: [ 'bar', 'baz' ] }
```

Reply:
Ah I see, I was worried we could end up with a situation here where they are not unique, but I guess that's actually impossible since those are always the full file paths right? In that case we can revert this change.

Reply:
Yes, full paths. We can keep the `merge` for now. The `Object.assign` was also due to performance reasons, but we can look into this when we get back to this part of the code at some point.